src/axom/quest/detail/shaping/shaping_helpers.hpp

```diff
@@ -224,6 +224,188 @@ void computeVolumeFractionsIdentity(mfem::DataCollection* dc,
+   * before querying them on the spatial index
+   *
+   * \note A projector callback must be supplied when \a FromDim is not equal
+   * to \a ToDim, the projector
+   * \note \a ToDim must be equal to \a DIM, the dimension of the spatial index
+   */
+template <int FromDim, int ToDim, typename InsideFunc>
+void sampleInOutField(const std::string shapeName,
+                      mfem::DataCollection* dc,
+                      shaping::QFunctionCollection& inoutQFuncs,
+                      int sampleRes,
+                      InsideFunc&& checkInside,
+                      PointProjector<FromDim, ToDim> projector = {})
+{
+  using FromPoint = primal::Point<double, FromDim>;
+  using ToPoint = primal::Point<double, ToDim>;
+  AXOM_ANNOTATE_SCOPE("sampleInOutField");
+
+  SLIC_ERROR_IF(FromDim != ToDim && !projector,
+                "A projector callback function is required when FromDim != ToDim");
+
+  auto* mesh = dc->GetMesh();
+  SLIC_ASSERT(mesh != nullptr);
+  const int NE = mesh->GetNE();
+  const int dim = mesh->Dimension();
+
+  // Generate a Quadrature Function with the geometric positions, if not already available
+  if(!inoutQFuncs.Has("positions"))
+  {
+    shaping::generatePositionsQFunction(mesh, inoutQFuncs, sampleRes);
+  }
+
+  // Access the positions QFunc and associated QuadratureSpace
+  mfem::QuadratureFunction* pos_coef = inoutQFuncs.Get("positions");
+  auto* sp = pos_coef->GetSpace();
+  const int nq = sp->GetIntRule(0).GetNPoints();
+
+  // Sample the in/out field at each point
+  // store in QField which we register with the QFunc collection
+  const std::string inoutName = axom::fmt::format("inout_{}", shapeName);
+  const int vdim = 1;
+  auto* inout = new mfem::QuadratureFunction(sp, vdim);
+  inoutQFuncs.Register(inoutName, inout, true);
+
+  mfem::DenseMatrix m;
+  mfem::Vector res;
+
+  axom::utilities::Timer timer(true);
+  for(int i = 0; i < NE; ++i)
+  {
+    pos_coef->GetValues(i, m);
+    inout->GetValues(i, res);
+
+    if(projector)
+    {
+      for(int p = 0; p < nq; ++p)
+      {
+        const ToPoint pt = projector(FromPoint(m.GetColumn(p), dim));
+        const bool in = checkInside(pt);
+        res(p) = in ? 1. : 0.;
+      }
+    }
+    else
+    {
+      for(int p = 0; p < nq; ++p)
+      {
+        const ToPoint pt(m.GetColumn(p), dim);
+        const bool in = checkInside(pt);
+        res(p) = in ? 1. : 0.;
+      }
+    }
+  }
+  timer.stop();
+
+  // print stats for rank 0
+  SLIC_INFO_ROOT(axom::fmt::format(
+    axom::utilities::locale(),
+    "\t Sampling inout field '{}' took {:.3Lf} seconds (@ {:L} queries per second)",
+    inoutName,
+    timer.elapsed(),
+    static_cast<int>((NE * nq) / timer.elapsed())));
+}
+
+/**
+   * \brief Samples the inout field over the indexed geometry, possibly using a
+   * callback function to project the input points (from the computational mesh)
+   * to query points on the spatial index
```

Comment:
Minor: This one doesn't have a projector function.

Perhaps it should (?), but I didn't add it.

Reply:
I added the projector plumbing for `computeVolumeFractionsBaseline` into `SamplingShaper` and the samplers.